### PR TITLE
Replace Path Naming by Operation Path Resolvers

### PR DESCRIPTION
**Pull request by Ener-Getick**
relates to https://github.com/api-platform/core/pull/666

### CODE REVIEW

**Comment by dunglas** [comment]
👍

**Comment by sroze** [comment]
Definitely 👍

File: core/operation-path-resolving.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+```yaml
+# app/config/config.yml
+
+api_platform:
+    default_operation_path_resolver: 'api_platform.operation_path_resolver.dash'
+```
+
+## Create a custom operation path resolver
+
+Let's assumes we need urls without separators (e.g. `api.tld/myresources`)
```
Comment (teohhanhui) [inline core/operation-path-resolving.md]: * assume
* URLs

File: core/operation-path-resolving.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Operation Path Resolving
+
+With API Platform Core, you can configure the root operation path resolver used to generate implicit paths.
```
Comment (teohhanhui) [inline core/operation-path-resolving.md]: you can configure the default resolver used to generate operation root paths.

File: core/operation-path-resolving.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# Operation Path Resolving
```
Comment (teohhanhui) [inline core/operation-path-resolving.md]: I suggest `Operation Path Naming` and `operation-path-naming.md`

File: core/operation-path-resolving.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+To change it to the dash resolver, add the following lines to `app/config/config.yml`:
+
+```yaml
+# app/config/config.yml
+
+api_platform:
+    default_operation_path_resolver: 'api_platform.operation_path_resolver.dash'
+```
+
+## Create a custom operation path resolver
```
Comment (teohhanhui) [inline core/operation-path-resolving.md]: Title Case

File: core/operation-path-resolving.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        $path .= '.{_format}';
+
+        return $path;
+    }
+}
+```
+
+Note that `$resourceShortName` contains a camel case string, by default the resource class name (e.g. `MyResource`).
+
+### Registering the service
```
Comment (teohhanhui) [inline core/operation-path-resolving.md]: Title Case

File: core/operation-path-resolving.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+<?xml version="1.0" encoding="UTF-8" ?>
+<services>
+    <service id="app.operation_path_resolver.no_separators" class="AppBundle\PathResolver\NoSeparatorsOperationPathResolver" public="false" />
+</services>
+```
+
+</configurations>
+
+### Configure it
```
Comment (teohhanhui) [inline core/operation-path-resolving.md]: Title Case

File: core/operation-path-resolving.md (hunk 2 above)
Comment (Ener-Getick) [inline core/operation-path-resolving.md]: That's not their root paths anymore that's their entire path.

File: core/operation-path-resolving.md (hunk 3 above)
Comment (Ener-Getick) [inline core/operation-path-resolving.md]: LGTM :+1:

**Comment by Ener-Getick** [comment]
Updated, thanks @teohhanhui :)

File: core/operation-path-resolving.md (hunk 2 above)
Comment (teohhanhui) [inline core/operation-path-resolving.md]: Noted. Remove the word root then :)

**Comment by dunglas** [comment]
Thank you @Ener-Getick

**Pull request merged**